### PR TITLE
chore: improve runfiles handling

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: examples/gem/lib/gem/BUILD [hunk 1]
```diff
@@ -32,3 +33,10 @@ rb_binary(
     args = ["lib/gem/version.rb"],
     deps = [":version"],
 )
+
+run_binary(
```
Comment (p0deje) [inline examples/gem/lib/gem/BUILD:37]: That's the first time I run into this rule. Is this a common way to run binary targets? I felt like `bazel run` cli is the default one and nobody normally uses binary targets as dependencies to other targets.

Comment (alexeagle) [reply examples/gem/lib/gem/BUILD:37]: They are for two very different use cases. Bazel run is for the side effects of running the program, run_binary is for using the program to produce outputs into bazel-out.

**Review by p0deje — APPROVED** [review]
LGTM and will happily merge as soon as CI passes.

**Comment by p0deje** [comment]
@alexeagle Do you know what's wrong with CI?

**Comment by p0deje** [comment]
Finished in #38

**Pull request closed without merging**